arcgis-ios-sdk-samples/Utility network/Create load report/CreateLoadReportViewController.swift

```diff
@@ -0,0 +1,357 @@
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["CreateLoadReportViewController"]
+        // Load the utility network and initialize properties.
+        loadUtilityNetwork()
+    }
+}
+
+// MARK: - UITableViewDelegate
+
+extension CreateLoadReportViewController: UITableViewDelegate, UITableViewDataSource {
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        Section.allCases[section].titleForHeader
+    }
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
+        Section.allCases.count
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch Section.allCases[section] {
+        case .included:
+            return includedPhases.count
+        case .excluded:
+            return excludedPhases.count
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: Section.allCases[indexPath.section].cellIdentifier, for: indexPath)
+        switch Section.allCases[indexPath.section] {
+        case .included:
+            let phase = includedPhases[indexPath.row]
+            cell.textLabel?.text = String(format: "Phase: %@", phase.name)
+            if let summary = summaries[phase] {
+                let formattedString = String(format: "C: %@  L: %@", numberFormatter.string(from: summary.totalCustomers as NSNumber)!, numberFormatter.string(from: summary.totalLoad as NSNumber)!)
+                cell.detailTextLabel?.text = formattedString
+            } else {
+                cell.detailTextLabel?.text = "N/A"
+            }
+        case .excluded:
+            cell.textLabel?.text = excludedPhases[indexPath.row].name
+        }
+        return cell
+    }
+    
+    func tableView(_ tableView: UITableView, editingStyleForRowAt indexPath: IndexPath) -> UITableViewCell.EditingStyle {
+        switch Section.allCases[indexPath.section] {
+        case .included:
+            return .delete
+        case .excluded:
+            return .insert
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
+        /// Binary search to find the insertion index in a sorted array.
+        func index(forInserting phase: AGSCodedValue, into array: [AGSCodedValue]) -> Int {
+            (array as NSArray).index(
+                of: phase,
+                inSortedRange: NSRange(array.indices),
+                options: [.insertionIndex],
+                usingComparator: { ($0 as! AGSCodedValue).name.compare(($1 as! AGSCodedValue).name) }
+            )
+        }
+        
+        switch editingStyle {
+        case .delete:
+            includedPhases.remove(at: indexPath.row)
+            let phase = includedPhases[indexPath.row]
```

Comment:
There a now some bugs in this sample. Rows do not have the right description when moving between sections and removing the last row in a section results in a crash. I think the issue is that the phase is being removed before the phase is retrieved from the array, both here and in the `insert` case.

Reply:
Oh my...  🤦 I was trying to be clever without even thinking... Fixed!